### ISSUE TITLE
in_array(): Argument #2 ($haystack) must be of type array, null given

### ISSUE THREAD

**Issue by cod3rshotout** [sebastienheyd/boilerplate#49]
Hi,

If you browse the route: `/admin/users`, you will get this error:

![Immagine 2022-01-26 115850](https://user-images.githubusercontent.com/72454584/151151443-a53c844c-6877-4ec3-bc0e-98389767777b.jpg)

which is caused by that line: `@if(in_array('filters', $datatable->buttons))`, in particular `UsersDatatable` doesn't have any buttons, so this will throw an exception, I temporary fixed this using:

```
$this->permissions('users_crud')
            ->locale([
                'deleteConfirm' => __('boilerplate::users.list.confirmdelete'),
                'deleteSuccess' => __('boilerplate::users.list.deletesuccess'),
            ])
            ->buttons([]) // MUST BE SPECIFIED
            ->order('created_at', 'desc')
            ->stateSave();
```

Also, I guess that this line:

`$userModel = config('boilerplate.laratrust.user.providers.users.model');`

should be:

`$userModel = config('boilerplate.laratrust.user');`

otherwise this will throw a type exception on ajax load

**Comment by sebastienheyd** [comment]
Hi,

Are you using the latest version ?

**Comment by cod3rshotout** [comment]
@sebastienheyd yes, also I noticed that since the latest version I doesn't have any filters on the DataTables columns

**Comment by sebastienheyd** [comment]
You don't have filters because you set `buttons([])`.

As you can see in [Datatable.php - ln. 34](https://github.com/sebastienheyd/boilerplate/blob/master/src/Datatables/Datatable.php#L34), `filters` is set by default.

Also in the latest version of [UsersDatatable.php - ln. 16](https://github.com/sebastienheyd/boilerplate/blob/master/src/Datatables/UsersDatatable.php#L16), `$userModel` is equal to `config('boilerplate.auth.providers.users.model')`

Are you sure you have the latest version? Also check your overloads if you have any.

**Comment by cod3rshotout** [comment]
> You don't have filters because you set `buttons([])`.
> 
> As you can see in [Datatable.php - ln. 34](https://github.com/sebastienheyd/boilerplate/blob/master/src/Datatables/Datatable.php#L34), `filters` is set by default.
> 
> Also in the latest version of [UsersDatatable.php - ln. 16](https://github.com/sebastienheyd/boilerplate/blob/master/src/Datatables/UsersDatatable.php#L16), `$userModel` is equal to `config('boilerplate.auth.providers.users.model')`
> 
> Are you sure you have the latest version? Also check your overloads if you have any.

For some reason that file wasn't updated.

When I load the user list I get this AJAX error:

MyApp\Boilerplate\Datatables\UsersDatatable::MyApp\Boilerplate\Datatables\{closure}(): Argument #1 ($user) must be of type MyApp\Boilerplate\Models\User, App\Models\Boilerplate\User given, called in /var/www/html/vendor/yajra/laravel-datatables-oracle/src/Utilities/Helper.php on line 65

how can I fix this?

**Comment by cod3rshotout** [comment]
> You don't have filters because you set `buttons([])`.
> 
> As you can see in [Datatable.php - ln. 34](https://github.com/sebastienheyd/boilerplate/blob/master/src/Datatables/Datatable.php#L34), `filters` is set by default.
> 
> Also in the latest version of [UsersDatatable.php - ln. 16](https://github.com/sebastienheyd/boilerplate/blob/master/src/Datatables/UsersDatatable.php#L16), `$userModel` is equal to `config('boilerplate.auth.providers.users.model')`
> 
> Are you sure you have the latest version? Also check your overloads if you have any.

I guess I found the problem about the `$userModel`, I've executed the `scaffhold` command to customize the boilerplate, but seems that some files like `UsersDatatable` aren't inserted in my solution (I don't know if it was wanted or was it an oversight?)

So in `UsersDatatable` I have:

```
use Sebastienheyd\Boilerplate\Models\User;

class UsersDatatable extends Datatable
{
    public $slug = 'users';

    public function datasource()
    {
        $userModel = config('boilerplate.auth.providers.users.model');
```

and in `myapp/config/boilerplate/auth` I have:

```
    'providers'     => [
        'users' => [
            'driver' => 'eloquent',
            'model'  => App\Models\Boilerplate\User::class,
            'table'  => 'users',
        ],
    ],
```

which is the model scaffholded (different from what the `UsersDatatable` class is loading). That's why I have this exception, I can fix this copying the `UsersDatatable` file in `myapp/app/Datatables`, but I suppose that if I call the scaffhold command this file should be automatically copied in there, agree with me?

**Comment by sebastienheyd** [comment]
Yes, you're right. Databables folder is not copied and it must be.

Ok, one more fix :)

**Comment by sebastienheyd** [comment]
Fixed in [7.11.4](https://github.com/sebastienheyd/boilerplate/releases/tag/7.11.4)
